File: compare_points/compare_points.py
import sys


def main(file_arg):
    with open(file_arg) as f:
        for line in f:
            O, P, Q, R = line.rstrip().split(' ')
            O = int(O)
            P = int(P)
            Q = int(Q)
            R = int(R)

            north = False
            south = False
            east = False
            west = False

            if (Q > O):
                east = True
            elif (Q < O):
                west = True

            if (P > R):
                south = True
            elif (P < R):
                north = True

            result = ''

            if (north):
                result += 'N'
            elif (south):
                result += 'S'

            if (east):
                result += 'E'
            elif (west):
                result += 'W'

            if not result:
                result = 'here'

            print(result)


if __name__ == '__main__':
    main(sys.argv[1])

File: compare_points/compare_points.js
var fs  = require("fs");

fs.readFileSync(process.argv[2]).toString().split('\n').forEach(function (line) {
    if (line != "") {
        inputs = line.split(' ');

        var O = parseInt(inputs[0]);
        var P = parseInt(inputs[1]);
        var Q = parseInt(inputs[2]);
        var R = parseInt(inputs[3]);

        var north = false;
        var south = false;
        var east = false;
        var west = false;

        if (Q > O) {
            east = true; 
        } else if (Q < O) {
            west = true; 
        }

        if (P > R) {
            south = true;
        } else if (P < R) {
            north = true;
        }

        result = ""

        if (north) {
            result += "N";
        } else if (south) {
            result += "S";
        }

        if (east) {
            result += "E";
        } else if (west) {
            result += "W";
        }

        if (!result) {
            result = "here";
        }

        console.log(result);
    }
});
